File: TeeNew/Builder2006/SeriesType_Arrow.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <stdio.h>
#pragma hdrstop

#include "SeriesType_Arrow.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TArrowSeriesForm *ArrowSeriesForm;
//---------------------------------------------------------------------------
__fastcall TArrowSeriesForm::TArrowSeriesForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TArrowSeriesForm::CheckBox3Click(TObject *Sender)
{
  Series1->ColorEachPoint = CheckBox3->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TArrowSeriesForm::FormCreate(TObject *Sender)
{
  Series1->ArrowWidth = 32;
  Series1->ArrowHeight = 24;
  Series1->XValues->DateTime = false;
  Series1->YValues->DateTime = false;
  AddRandomArrows();
}
//---------------------------------------------------------------------------
void __fastcall TArrowSeriesForm::AddRandomArrows()
{
  double x0,y0,x1,y1;
  Series1->Clear();
  for (int t=1; t<41; t++)
  {
    x0 = random(1000);
    y0 = random(1000);

    x1 = random(300) - 150.0;
    if (x1<50.0) x1 = 50.0;
    x1 = x1+x0;

    y1 = random(300) - 150.0;
    if (y1<50.0) y1 = 50.0;
    y1 = y1+y0;

    Series1->AddArrow(x0,y0,x1,y1, "", clTeeColor );
  }
}
//---------------------------------------------------------------------------

void __fastcall TArrowSeriesForm::CheckBox1Click(TObject *Sender)
{
  Timer1->Enabled = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TArrowSeriesForm::CheckBox2Click(TObject *Sender)
{
  Chart1->View3D = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TArrowSeriesForm::Timer1Timer(TObject *Sender)
{
  Timer1->Enabled = false;
  for (int t=0; t< Series1->Count(); t++)
  {
      Series1->StartXValues->Value[t] = Series1->StartXValues->Value[t]+random(100)-50.0;
      Series1->StartYValues->Value[t]= Series1->StartYValues->Value[t]+random(100)-50.0;
      Series1->EndXValues->Value[t]  = Series1->EndXValues->Value[t]+random(100)-50.0;
      Series1->EndYValues->Value[t]  = Series1->EndYValues->Value[t]+random(100)-50.0;
  }
  Series1->Repaint();
  Timer1->Enabled = true;
}
//---------------------------------------------------------------------------

File: TeeNew/SeriesType_Arrow.pas
unit SeriesType_Arrow;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series, TeeTools, ArrowCha, TeeGDIPlus;

type
  TArrowSeriesForm = class(TBaseForm)
    CheckBox3: TCheckBox;
    Series1: TArrowSeries;
    CheckBox1: TCheckBox;
    Timer1: TTimer;
    CheckBox2: TCheckBox;
    procedure CheckBox3Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure Timer1Timer(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
   procedure AddRandomArrows;
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TArrowSeriesForm.CheckBox3Click(Sender: TObject);
begin
  Series1.ColorEachPoint:=CheckBox3.Checked;
end;

procedure TArrowSeriesForm.FormCreate(Sender: TObject);
begin
  inherited;
  With Series1 do
  Begin
    ArrowWidth:=32;
    ArrowHeight:=24;
    XValues.DateTime:=False;
    YValues.DateTime:=False;
  end;
  AddRandomArrows;
end;

procedure TArrowSeriesForm.AddRandomArrows;
var x0,y0,x1,y1:Double;
    t:Integer;
begin
  With Series1 do
  Begin
    Clear;
    for t:=1 to 40 do
    begin
      x0:=Random( 1000 );
      y0:=Random( 1000 );

      x1:=Random( 300 ) - 150.0;
      if x1<50 then x1:=50;
      x1:=x1+x0;

      y1:=Random( 300 ) - 150.0;
      if y1<50 then y1:=50;
      y1:=y1+y0;

      AddArrow( x0,y0,x1,y1, '', clTeeColor );
    end;
  end;
end;

procedure TArrowSeriesForm.CheckBox1Click(Sender: TObject);
begin
  Timer1.Enabled:=CheckBox1.Checked;
end;

procedure TArrowSeriesForm.Timer1Timer(Sender: TObject);
var t:Integer;
begin
  Timer1.Enabled:=False;
  With Series1 do
  Begin
    for t:=0 to Count-1 do
    Begin
      StartXValues[t]:=StartXValues[t]+Random(100)-50.0;
      StartYValues[t]:=StartYValues[t]+Random(100)-50.0;
      EndXValues[t]  :=EndXValues[t]+Random(100)-50.0;
      EndYValues[t]  :=EndYValues[t]+Random(100)-50.0;
    End;
    Repaint;
  End;
  Timer1.Enabled:=True;
end;

procedure TArrowSeriesForm.CheckBox2Click(Sender: TObject);
begin
  Chart1.View3D:=CheckBox2.Checked
end;

initialization
  RegisterClass(TArrowSeriesForm);
end.
